test/grpc_web_server.dart

```diff
@@ -164,4 +168,54 @@ if you are running tests locally.
     tempDir.deleteSync(recursive: true);
   }
   channel.sink.add('EXITED');
+
+  await server.shutdown().catchError((_) {});
+  await httpServer.close().catchError((_) {});
+}
+
+final testCases = <String, void Function(HttpResponse)>{
+  'test:cors': (rs) {
+    rs.headers.removeAll('Access-Control-Allow-Origin');
+    rs.headers.add(HttpHeaders.contentTypeHeader, 'text/html');
+    rs.write('some body');
+    rs.close();
+  },
+  'test:status-503': (rs) {
+    rs.headers.add(HttpHeaders.contentTypeHeader, 'text/html');
+    rs.statusCode = HttpStatus.serviceUnavailable;
+    rs.write('some body');
+    rs.close();
+  },
+  'test:bad-content-type': (rs) {
+    rs.headers.add(HttpHeaders.contentTypeHeader, 'text/html');
+    rs.statusCode = HttpStatus.ok;
+    rs.write('some body');
+    rs.close();
+  },
+};
+
+void defaultHandler(HttpResponse rs) {
+  rs.close();
+}
+
+Future<HttpServer> startHttpServer() async {
+  final server = await HttpServer.bind(InternetAddress.loopbackIPv4, 51428);
```

Comment:
Could you avoid hard-coding the port?

Reply:
Good catch. This was left-over from debugging.